hazelcast/src/main/java/com/hazelcast/internal/config/ConfigXmlGeneratorHelper.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.config;
+
+import com.hazelcast.config.CompactSerializationConfig;
+import com.hazelcast.config.ConfigXmlGenerator.XmlGenerator;
+import com.hazelcast.internal.util.MapUtil;
+import com.hazelcast.internal.util.TriTuple;
+import com.hazelcast.nio.serialization.compact.CompactSerializer;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.hazelcast.config.CompactSerializationConfigAccessor.getCompactSerializableClassNames;
+import static com.hazelcast.config.CompactSerializationConfigAccessor.getRegistrations;
+import static com.hazelcast.config.CompactSerializationConfigAccessor.getSerializerClassNames;
+
+/**
+ * Contains utility methods to be used in the client and member XML config
+ * generators.
+ */
+public final class ConfigXmlGeneratorHelper {
+
+    private ConfigXmlGeneratorHelper() {
+    }
+
+
+    /**
+     * Generates the Compact serialization configuration out of the given
+     * configuration.
+     */
+    public static void compactSerialization(XmlGenerator gen, CompactSerializationConfig config) {
+        if (!config.isEnabled()) {
+            // We don't want to output any configuration if it is not enabled by
+            // default during the BETA period. This check should be removed
```

Comment:
why don't we remove this in 5.2-BETA?

Reply:
because we haven't merged all the PRs. once we do it for 5.2-BETA, I will remove this, along with other places

Reply:
will you do it for 5.2-BETA release? I think we should.

Reply:
Yes, we should. I was planning to do it after merging these PRs + schema distribution PR